### PR TITLE
Implement SoundAgent (#45)

### PR DESCRIPTION
**Pull request by tommy5oo**

### CODE REVIEW

File: nugu-interface/src/main/java/com/skt/nugu/sdk/core/interfaces/mediaplayer/PlayerFactory.kt [hunk 1]
```diff
@@ -43,4 +43,10 @@ interface PlayerFactory {
      * Create a player for playing ringtone
      */
 //    fun createRingtonePlayer(): MediaPlayerInterface
+
+    /**
+     * Create a player for playing beep
+     */
+    fun createBeepPlayer(): MediaPlayerInterface
```
Comment (shshshl09) [inline nugu-interface/src/main/java/com/skt/nugu/sdk/core/interfaces/mediaplayer/PlayerFactory.kt]: Maybe enough MediaPlayerInterface -> UriSourcePlayablePlayer

File: nugu-interface/src/main/java/com/skt/nugu/sdk/core/interfaces/capability/sound/SoundProvider.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+package com.skt.nugu.sdk.core.interfaces.capability.sound
+
+import java.net.URI
+
+
+/**
+ * The public interface for SoundAgent
+ */
+interface SoundProvider {
+    fun getContentUri(beepName: String) : URI
```
Comment (shshshl09) [inline nugu-interface/src/main/java/com/skt/nugu/sdk/core/interfaces/capability/sound/SoundProvider.kt]: I think beepName defined by enum.

File: nugu-core/src/main/java/com/skt/nugu/sdk/core/capabilityagents/impl/DefaultSoundAgent.kt [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+            sendEvent(NAME_BEEP_SUCCEEDED, playServiceId)
+        }
+
+        private fun sendBeepFailedEvent(playServiceId: String) {
+            sendEvent(NAME_BEEP_FAILED, playServiceId)
+        }
+
+        private fun sendEvent(name: String, playServiceId: String) {
+            Logger.d(TAG, "[sendEvent] name: $name, playServiceId: $playServiceId")
+            contextManager.getContext(object : ContextRequester {
```
Comment (shshshl09) [inline nugu-core/src/main/java/com/skt/nugu/sdk/core/capabilityagents/impl/DefaultSoundAgent.kt]: Use  contextManager.getContextWithoutUpdate

File: nugu-interface/src/main/java/com/skt/nugu/sdk/core/interfaces/capability/sound/SoundProvider.kt (hunk 2 above)
Comment (tommy5oo) [reply nugu-interface/src/main/java/com/skt/nugu/sdk/core/interfaces/capability/sound/SoundProvider.kt]: Done

File: nugu-interface/src/main/java/com/skt/nugu/sdk/core/interfaces/mediaplayer/PlayerFactory.kt (hunk 1 above)
Comment (tommy5oo) [reply nugu-interface/src/main/java/com/skt/nugu/sdk/core/interfaces/mediaplayer/PlayerFactory.kt]: Done

File: nugu-core/src/main/java/com/skt/nugu/sdk/core/capabilityagents/impl/DefaultSoundAgent.kt (hunk 3 above)
Comment (tommy5oo) [reply nugu-core/src/main/java/com/skt/nugu/sdk/core/capabilityagents/impl/DefaultSoundAgent.kt]: Done!

File: nugu-interface/src/main/java/com/skt/nugu/sdk/core/interfaces/capability/sound/SoundProvider.kt (hunk 2 above)
Comment (tommy5oo) [reply nugu-interface/src/main/java/com/skt/nugu/sdk/core/interfaces/capability/sound/SoundProvider.kt]: Done

File: nugu-client-kit/src/main/java/com/skt/nugu/sdk/client/NuguClient.kt [hunk 4]
```diff
@@ -134,6 +137,7 @@ class NuguClient private constructor(
         internal var extensionAgentFactory: ExtensionAgentFactory = DefaultExtensionAgent.FACTORY
         internal var displayAgentFactory: DisplayAgentFactory? = DefaultDisplayAgent.FACTORY
         internal var locationAgentFactory: LocationAgentFactory? = DefaultLocationAgent.FACTORY
+        internal var soundAgentFactory: SoundAgentFactory = DefaultSoundAgent.FACTORY
```
Comment (shshshl09) [inline nugu-client-kit/src/main/java/com/skt/nugu/sdk/client/NuguClient.kt]: should be nullable

**Pull request merged**